runtime/cudaq/platform/default/rest_server/helpers/RestRemoteClient.cpp

```diff
@@ -0,0 +1,223 @@
+
+#include "JsonConvert.h"
+#include "common/Logger.h"
+#include "common/RemoteKernelExecutor.h"
+#include "common/RestClient.h"
+#include "cudaq.h"
+
+namespace {
+using namespace mlir;
+class RemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
+  std::string m_url;
+  static inline const std::vector<std::string> clientPasses = {
+      "func.func(unwind-lowering)",
+      "func.func(indirect-to-direct-calls)",
+      "inline",
+      "canonicalize",
+      "apply-op-specialization",
+      "func.func(memtoreg{quantum=0})",
+      "canonicalize",
+      "expand-measurements",
+      "cc-loop-normalize",
+      "cc-loop-unroll",
+      "canonicalize",
+      "func.func(add-dealloc)",
+      "func.func(quake-add-metadata)",
+      "canonicalize",
+      "func.func(lower-to-cfg)",
+      "func.func(combine-quantum-alloc)",
+      "canonicalize",
+      "cse",
+      "quake-to-qir"};
+  static inline const std::vector<std::string> serverPasses = {};
+
+public:
+  virtual void setConfig(
+      const std::unordered_map<std::string, std::string> &configs) override {
+    const auto urlIter = configs.find("url");
+    if (urlIter != configs.end())
+      m_url = urlIter->second;
+  }
+
+  std::string constructKernelPayload(MLIRContext &mlirContext,
+                                     const std::string &name,
+                                     void (*kernelFunc)(void *), void *args,
+                                     std::uint64_t voidStarSize) {
+    if (cudaq::__internal__::isLibraryMode(name)) {
+      // Library mode: retrieve the embedded bitcode in the executable.
+      const auto path = llvm::sys::fs::getMainExecutable(nullptr, nullptr);
+      // Load the object file
+      auto [objBin, objBuffer] =
+          llvm::cantFail(llvm::object::ObjectFile::createObjectFile(path))
+              .takeBinary();
+      if (!objBin)
+        throw std::runtime_error("Failed to load binary object file");
+      for (const auto &section : objBin->sections()) {
+        // Get the bitcode section
+        if (section.isBitcode()) {
+          llvm::MemoryBufferRef llvmBc(llvm::cantFail(section.getContents()),
+                                       "Bitcode");
+          return llvm::encodeBase64(llvmBc.getBuffer());
+        }
+      }
+      return "";
+    } else {
+      // Get the quake representation of the kernel
+      auto quakeCode = cudaq::get_quake_by_name(name);
+      auto module = parseSourceString<ModuleOp>(quakeCode, &mlirContext);
+      if (!module)
+        throw std::runtime_error("module cannot be parsed");
+
+      // Extract the kernel name
+      auto func = module->lookupSymbol<mlir::func::FuncOp>(
+          std::string("__nvqpp__mlirgen__") + name);
+
+      // Create a new Module to clone the function into
+      auto location = FileLineColLoc::get(&mlirContext, "<builder>", 1, 1);
+      ImplicitLocOpBuilder builder(location, &mlirContext);
+
+      // FIXME this should be added to the builder.
```

Comment:
Is this FIXME for now or for later? If later, we should probably create an issue to track it.

Reply:
This is not relevant here. Removed in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/fc19b918ee2e26b2ca2d137a7dfe6fc5549b1e62